editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/maven/CamelProjectConfigurator.java

```diff
@@ -93,9 +101,56 @@ public void configure(ProjectConfigurationRequest request, IProgressMonitor moni
 				// enable the camel nature
 				configureNature(request.getProject(), request.getMavenProject(), monitor);
 			}
+			// handle linked resources for WAR deployments
+			if (isWARProject(request.getProject())) {
+				configureWARStructureMapping(request.getProject());	
+			}		
 		}
 	}
+	
+	private void configureWARStructureMapping(IProject project) throws CoreException {
+		final IVirtualComponent c = ComponentCore.createComponent(project, false);
+		c.create(0, null);
+		final IVirtualFolder webroot = c.getRootFolder();
+		final IVirtualFolder classesFolder = webroot.getFolder("/WEB-INF/classes"); //$NON-NLS-1$
+		IMavenProjectFacade m2prj = MavenPlugin.getMavenProjectRegistry().create(project, new NullProgressMonitor());
+		updateMappings(m2prj.getCompileSourceLocations(), project, classesFolder);
+		updateMappings(m2prj.getTestCompileSourceLocations(), project, classesFolder);
+	}
+	
+	/**
+	 * this methods maps a given set of local paths to a path on runtime / in the WAR
+	 * 
+	 * @param paths
+	 * @param project
+	 * @param vFolder
+	 * @throws CoreException
+	 */
+	private void updateMappings(IPath[] paths, IProject project, IVirtualFolder vFolder) throws CoreException {
+		for (IPath sourceLoc : paths) {
+			IFolder srcFolder = project.getFolder(sourceLoc);
+			IVirtualResource[] mappings = ComponentCore.createResources(srcFolder);
+			boolean found = false;
+			for (IVirtualResource mapping : mappings) {
+				if (mapping.getProjectRelativePath().toOSString().equals("/" + srcFolder.getProjectRelativePath().toOSString())) {
+					mapping.createLink(new Path("/").append(sourceLoc), 0, null);
+					found = true;
+					break;
+				}
+			}
+			if (!found) {
+				//create link for source folder only when it is not mapped
+				vFolder.createLink(new Path("/").append(sourceLoc), 0, null);
+			}
+		}
+	}
+	
+	private boolean isWARProject(IProject project) {
+		IMavenProjectFacade m2prj = MavenPlugin.getMavenProjectRegistry().create(project, new NullProgressMonitor());
+		return m2prj.getPackaging().equalsIgnoreCase(CamelProjectConfigurator.WAR_PACKAGE);
```

Comment:
invert equals to avoid NPE in case the project is malformed and have no packaging defined